### DISCUSSION TITLE
Plans for MISRA compliance for CFS?

### DISCUSSION THREAD

**Discussion by shuq0** [nasa/cFS#449] [Ideas]
Are there any plans from the core team to harden the CFS code base to be MISRA compliant? 
https://www.misra.org.uk/app/uploads/2021/06/MISRA-Compliance-2020.pdf

**Comment by skliper** [comment]
There is a general goal of becoming more compliant with industry standards.  There are trades related to competing standards, conflicting rules, impacts to historical/heritage software, performance impacts, etc.  At this point we don't have resources dedicated specifically towards an major MISRA compliance effort, but the intent is to make progress in that direction as opportunities arise (where the trades justify it).  Note we have been experimenting with adding ruleset compliance checks in CI (https://github.com/nasa/cFS/blob/main/.github/codeql/jpl-misra.qls, https://github.com/nasa/cFS/blob/main/.github/workflows/static-analysis-misra.yml)

**Reply by shuq0** [reply]
Thanks for the links. Its a good starting point!
